### PR TITLE
Packages for RG350 and GCW0 

### PR DESCRIPTION
**Pull request by glebm**
~~Known issue: no audio in movies~~

Since these devices have SDL 2.0.3 and almost the same buttons as the Switch, we should be able to use the same control scheme here.

Also fixes the RS90 package (I don't have an RS90 to test with but clearly I was using the wrong package format).

### CODE REVIEW

**Comment by glebm** [comment]
When combined with the Switch branch (https://github.com/glebm/devilutionX/tree/rg350-controls-switch) we get this:

* Short video: https://imgur.com/gallery/65P94YK
* OPK package: [devilutionx-rg350.opk.zip](https://github.com/diasurgical/devilutionX/files/3727224/devilutionx-rg350.opk.zip)

File: SourceX/dx.cpp [hunk 1]
```diff
@@ -61,9 +61,7 @@ void dx_create_primary_surface()
 #ifndef USE_SDL1
 	if (renderer) {
 		int width, height;
-		if (SDL_GetRendererOutputSize(renderer, &width, &height) <= -1) {
-			ErrSdl();
-		}
+		SDL_RenderGetLogicalSize(renderer, &width, &height);
```
Comment (glebm) [inline SourceX/dx.cpp]: @AJenbo This is an important fix. Here is how the game looked without it:

IMG_20191014_234813

(output size here is 320x240, logical size 640x480)

**Comment by glebm** [comment]
Latest commit fixes audio playback in movies!

Looks like the `AudioDevice` implementation is a bit buggy on SDL 2.0.3, so we now simply use the legacy non-device one there.

**Comment by glebm** [comment]
@AJenbo Can this be merged? It doesn't touch any files outside of Packaging/OpenDingux except some general fixes.

**Comment by guillaume-roche** [comment]
@glebm this is really cool to see it running on the RG350 and GCW0 handhelds! Do you know if it would be feasible to make it support also OpenDingux for the Dingoo A320? I own one, so I can help supporting it.

File: Packaging/OpenDingux/build-rg350.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	prepare_buildroot
+	build
+	package
+}
+
+prepare_buildroot() {
+	if [[ -d $BUILDROOT ]]; then
+		return
+	fi
+	git clone --depth=1 -b backport-glibc-host-fix https://github.com/glebm/RG350_buildroot.git "$BUILDROOT"
```
Comment (AJenbo) [inline Packaging/OpenDingux/build-rg350.sh]: Why not use https://github.com/tonyjih/RG350_buildroot It doesn't appear that your fork has any commits on top of it as is already lacking a bit behind?

File: Packaging/OpenDingux/buildroot_rg350_defconfig [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+BR2_OPTIMIZE_2=y
+BR2_KERNEL_HEADERS_3_12=y
+BR2_TOOLCHAIN_BUILDROOT_LARGEFILE=y
+BR2_TOOLCHAIN_BUILDROOT_INET_IPV6=y
+BR2_TOOLCHAIN_BUILDROOT_WCHAR=y
+BR2_BINUTILS_EXTRA_CONFIG_OPTIONS="--enable-lto"
+BR2_GCC_VERSION_4_9_X=y
+BR2_EXTRA_GCC_CONFIG_OPTIONS="--enable-lto"
+BR2_TOOLCHAIN_BUILDROOT_CXX=y
+BR2_PACKAGE_SDL2_IMAGE=y
```
Comment (AJenbo) [inline Packaging/OpenDingux/buildroot_rg350_defconfig]: I don't know exactly what this does, but SDL2_image isn't required by the project.

File: Packaging/OpenDingux/package-opk.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+readonly IN="${2:-../../build/devilutionx.dge}"
+
+readonly TMP="tmp/opk"
+
+echo 1>&2 Packaging ${OUT}...
+
+set -x
+rm -rf "${TMP}"
+mkdir -p "${TMP}"
+cat > "${TMP}/default.gcw0.desktop" <<EOF
```
Comment (AJenbo) [inline Packaging/OpenDingux/package-opk.sh]: We should probably think about making a general .desktop template for all unix systems

File: Packaging/OpenDingux/buildroot_rg350_defconfig (hunk 3 above)
Comment (glebm) [reply Packaging/OpenDingux/buildroot_rg350_defconfig]: It is required by the buildroot, they currently compile SDL2 expecting IMAGE to be there.

File: Packaging/OpenDingux/package-opk.sh (hunk 4 above)
Comment (glebm) [reply Packaging/OpenDingux/package-opk.sh]: This is rather specific to Dingux. There is a custom manual and a custom flag to enable the controller.

File: Packaging/OpenDingux/build-rg350.sh (hunk 2 above)
Comment (glebm) [reply Packaging/OpenDingux/build-rg350.sh]: My PR to it was just merged today :)

File: README.md [hunk 5]
```diff
@@ -151,22 +151,34 @@ The build script does the following:
...
+For OpenDingux builds `mksquashfs` needs to be installed.
 
-The OpenDingux build uses the buildroot at `$HOME/buildroot-2018.02.9-opendingux-musl`.
+### RetroFW (RS97, RG300, LDK)
+
+The RetroFW build uses the buildroot at `$HOME/buildroot-2018.02.9-retrofw`.
 
 ~~~ bash
-Packaging/OpenDingux/build-opendingux-sdl1.sh
+Packaging/OpenDingux/build-retrofw.sh
```
Comment (AJenbo) [inline README.md]: Maybe a better folder/title name would be RetroGamerConsols/Retro Gamer Consols as that appears to be the common denominator for all of these devices.

File: Packaging/OpenDingux/build-rg350.sh (hunk 2 above)
Comment (glebm) [reply Packaging/OpenDingux/build-rg350.sh]: Done

File: Packaging/OpenDingux/buildroot_rg350_defconfig (hunk 3 above)
Comment (glebm) [reply Packaging/OpenDingux/buildroot_rg350_defconfig]: We could try to customize it but I think the effort is just not worth it at this point.

File: Packaging/OpenDingux/package-opk.sh (hunk 4 above)
Comment (AJenbo) [reply Packaging/OpenDingux/package-opk.sh]: Maybe they could be added to the general one, not saying we have to do this now, but maybe good to keep in mind.

File: README.md (hunk 5 above)
Comment (glebm) [reply README.md]: I'd like to not rename it yet. I have a number of commits in the controller supports branch on top of it and it'd be nice to minimize merge conflicts when we eventually add controller support to mainline.

File: Packaging/OpenDingux/package-opk.sh (hunk 4 above)
Comment (glebm) [reply Packaging/OpenDingux/package-opk.sh]: There is also a custom launch command to add SDL2 controller mappings,
so it's not just additive:

https://github.com/glebm/devilutionX/blob/7ad53fff896433e56cbcd18858f8bda8395adb72/Packaging/OpenDingux/package-opk.sh#L20-L30

But yeah maybe a template in the future

File: README.md (hunk 5 above)
Comment (glebm) [reply README.md]: Besides, RetroFW is a fork of OpenDingux

**Comment by glebm** [comment]
> Do you know if it would be feasible to also support OpenDingux for the Dingoo A320?

@guillaume-roche I don't see why not :)

File: README.md (hunk 5 above)
Comment (AJenbo) [reply README.md]: Sounds good thanks for the clarification

**Comment by AJenbo** [comment]
Merged manually: https://github.com/diasurgical/devilutionX/commit/afaa25193ab27442565134f9bb16d3bc59858e55 https://github.com/diasurgical/devilutionX/commit/56b77eb6337ae8ab5d2fcb379691cf9a33957e29 https://github.com/diasurgical/devilutionX/commit/f13d2d193af5959c0d0294d248c4566b12744607 https://github.com/diasurgical/devilutionX/commit/f0eab356728a98f1799ab3a9a098ceca250ca08b

**Pull request closed without merging**